torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -55,6 +55,24 @@ struct TORCH_API AccumulateGrad : public Node {
         // holding references to the Tensor and that is fine since these are not
         // exposed to the user.
         update_grad(new_grad_copy.detach());
+      } else if (
+          !GradMode::is_enabled() && new_grad_copy.is_sparse() &&
+          new_grad_copy._indices().is_contiguous() &&
+          new_grad_copy._values().is_contiguous() &&
+          new_grad_use_count <= num_expected_refs) {
```

Comment:
Actually following the comment above. We want to check the use_count for the indices and values as well here no?